x/ccv/provider/keeper/relay.go

```diff
@@ -278,3 +296,28 @@ func (k Keeper) HandleSlashPacket(ctx sdk.Context, chainID string, data ccv.Slas
 
 	return true, nil
 }
+
+// EndBlockCIS contains the EndBlock logic needed for
+// the Consumer Chain Removal sub-protocol
+func (k Keeper) EndBlockCCR(ctx sdk.Context) {
+	currentTime := uint64(ctx.BlockTime().UnixNano())
+
+	// iterate over initTimeoutTimestamps
+	var removedChainIds []string
+	k.IterateInitTimeoutTimestamp(ctx, func(chainID string, ts uint64) bool {
+		if currentTime > ts {
```

Comment:
Would it be better to only store the timestamp when initialization started, and compute the timeout timestamp here with the most up to date `InitTimeoutPeriod` ? 

Maybe the desired functionality is that the new timeout period is effective immediately (even for in progress initializations), but I'm not sure

Reply:
Not sure that I follow. Do you have a specific concern in mind? What exactly do you mean by "when initialization started"? What initialization? IIUC what you're suggesting, it requires to compute the timeout timestamp in every EndBlock, right?

Reply:
The edge case I'm referring to is that a governance proposal changes the `InitTimeoutPeriod` param in the midst of one or more in-progress channel initializations. If you compute timeout timestamps during the endblock routine instead of precomputing the timeout timestamp as you've implemented here, you'd always be timing out channel initializations with the most recent `InitTimeoutPeriod`. 

Note that during the endblock, you do not have to compute a timeout timestamp for every in progress initialization. You can store the timestamps in order and stop iteration once you've reached the first chain that doesn't need a timeout

Reply:
I understand. It depends on the desired outcome. Do we want for a governance proposal to affect the init timeout timestamp of consumers with the channel handshake in progress? @jtremback 

The same question is relevant for the VSC timeout also (see https://github.com/cosmos/interchain-security/issues/283). Do we want a governance proposal to change how long we are waiting for the maturity of already sent VSCPackets?

Reply:
After discussing with @jtremback we decided to store the timeout timestamp instead of the timestamp of starting the channel handshake. This will be consistent of how the `UnbondingPeriod` is used in the SDK (when computing the completion time of unbonding operations).